Fri Sep  8 20:05:09 CEST 2006  AlexExtreme <alex@alex-smith.me.uk>
  * libfwnetconfig: reverted to dhcpcd as dhcp client, dhclient isn't working right and we need this to work flawlessly for 0.5 ;)
diff -rN -u old-frugalwareutils/libfwnetconfig/libfwnetconfig.c new-frugalwareutils/libfwnetconfig/libfwnetconfig.c
--- old-frugalwareutils/libfwnetconfig/libfwnetconfig.c	2006-09-08 19:34:24.000000000 +0200
+++ new-frugalwareutils/libfwnetconfig/libfwnetconfig.c	2006-09-08 19:34:25.000000000 +0200
@@ -229,10 +229,10 @@
 	if(dhcp)
 	{
 		char line[7];
-		if (!strcmp("dhcpcd", iface->dhcpclient)) {
-			ptr = g_strdup_printf("/etc/dhcpc/dhcpcd-%s.pid", iface->name);
-		} else
+		if (!strcmp("dhclient", iface->dhcpclient)) {
 			ptr = g_strdup_printf("/var/run/dhclient-%s.pid", iface->name);
+		} else
+			ptr = g_strdup_printf("/etc/dhcpc/dhcpcd-%s.pid", iface->name);
 		fp = fopen(ptr, "r");
 		FWUTIL_FREE(ptr);
 		if(fp != NULL)
@@ -245,8 +245,8 @@
 			else if (i>0)
 				printf("kill(%d, 15);\n", i);
 			
-			// if we're not using dhcpcd...
-			if (strcmp("dhcpcd", iface->dhcpclient)) {
+			// if we're using dhclient...
+			if (!strcmp("dhclient", iface->dhcpclient)) {
 				ptr = g_strdup_printf("ifconfig %s down", iface->name);
 				fwutil_system(ptr);
 				FWUTIL_FREE(ptr);
@@ -376,13 +376,13 @@
 	// set up the interface
 	if(dhcp)
 	{
-		if (!strcmp(iface->dhcpclient, "dhcpcd"))
+		if (!strcmp(iface->dhcpclient, "dhclient"))
+			ptr = g_strdup_printf("dhclient -pf /var/run/dhclient-%s.pid -lf /var/lib/dhclient/dhclient-%s.leases -q %s", iface->name, iface->name, iface->name);
+		else {
 			if(strlen(iface->dhcp_opts))
 				ptr = g_strdup_printf("dhcpcd %s %s", iface->dhcp_opts, iface->name);
 			else
 				ptr = g_strdup_printf("dhcpcd -t 10 %s", iface->name);
-		else {
-			ptr = g_strdup_printf("dhclient -pf /var/run/dhclient-%s.pid -lf /var/lib/dhclient/dhclient-%s.leases -q %s", iface->name, iface->name, iface->name);
 		}
 		
 		ret += fwutil_system(ptr);

